tidy3d/components/device/simulation.py

```diff
@@ -958,3 +1086,53 @@ def from_scene(cls, scene: Scene, **kwargs) -> HeatSimulation:
             medium=scene.medium,
             **kwargs,
         )
+
+    def _get_simulation_types(self) -> list[DeviceSimulationType]:
+        """
+        Checks through BCs and sources and returns the
+        types of simulations.
+        """
+        simulation_types = []
+
+        heat_source_present = False
+        # check for heat simulation
+        for s in self.sources:
+            if isinstance(s, HeatSourceTypes):
+                heat_source_present = True
+
+        heat_BCs_present = False
+        for bc in self.boundary_spec:
+            if isinstance(bc.condition, HeatBCTypes):
+                heat_BCs_present = True
+
+        if heat_source_present and not heat_BCs_present:
+            raise SetupError("Heat sources defined but no heat BCs present.")
+        elif heat_BCs_present or heat_source_present:
+            simulation_types.append(DeviceSimulationType.HEAT)
+
+        # check for conduction simulation
+        electric_spec_present = False
+        for structure in self.structures:
+            if structure.medium.electric_spec is not None:
+                electric_spec_present = True
+
+        electric_BCs_present = False
+        for bc in self.boundary_spec:
+            if isinstance(bc.condition, ElectricBCTypes):
+                electric_BCs_present = True
+
+        if electric_BCs_present and not electric_spec_present:
+            raise SetupError(
+                "Electric BC were specified but not all"
+                " structures have materials with electric"
```

Comment:
it seems like it's more like **no** structures have materials with electric spec?

also, might be helpful to be more specific and point to the field they need to set 
```
not all structures have a '.medium' with `.electric_spec` present
```